vulcanai2/models/basenetwork.py

```diff
@@ -261,6 +271,14 @@ def get_weights(self):
         """
         return self.state_dict()
 
+    def print_model_structure(self, input_dim):
```

Comment:
actually, since this is a part of basenetwork, do we store the input dimensions anywhere in this class? perhaps we can default to that.

Reply:
yeah good idea!